orttraining/orttraining/python/training/_ortmodule_graph_execution_manager.py

```diff
@@ -61,6 +61,7 @@ def _run_forward(execution_session, onnx_model, device, *inputs, **kwargs):
 
     output_info = [(output.shape, output.device, output.dtype) for output in user_outputs]
     run_info = onnxruntime.training.RunStateInfo(state, output_info)
+    del state
```

Comment:
Does this statement have any effect?

As far as I can tell, you're incrementing the refcount on `state` when passing it to the `RunStateInfo` initializer and storing it in `run_info`. The refcount from the local variable `state` will be dropped on the next line when the function returns.

So I think this is a no-op... but if it isn't please add a comment explaining the (non-obvious, to me at least!) behavior here.

Reply:
@mrry You are right, it's likely a no-op, based on my reading last night it seems python decrements the reference count of variables that go out of scope, I had this feeling too but put this out of abundance of caution and some paranoia ... I'll remove it.